### DISCUSSION TITLE
How to assure volume being rebuilt, if it is rarely attached?

### DISCUSSION THREAD

**Discussion by Szwendacz99** [longhorn/longhorn#9770] [Q&A] [answered]
I have volume that is attached for short time one a day. How to handle rebuilding of such volumes automatially, as they dont do that, and they cannot rebuild during that short time of being attached. This is more serious than it looks, as I got such volume twice in infinite attaching, which required from me deleting and restoring from backup.

**Comment by derekbit** [comment]
Do you have recurring jobs for the volume?

**Reply by Szwendacz99** [reply]
Yes, snapsots and backups

**Reply by derekbit** [reply]
Why not manually attach them, wait for finishing rebuilding and detach?

**Reply by Szwendacz99** [reply]
Seem like a lot of wasted time. Also would require some external monitoring automatization to not miss such volumes.

**Reply by derekbit** [reply, marked as answer]
We have a feature ticket for the offline rebuilding. Although it is for v2 volume, it can be extended to v1 volume as well.
https://github.com/longhorn/longhorn/issues/8443
What you need is it, correct?

**Reply by Szwendacz99** [reply]
Yes, it seems like something that would solve the problem. If v2 volumes are expected to be the next default, I guess v2 would solve the problem as well.
